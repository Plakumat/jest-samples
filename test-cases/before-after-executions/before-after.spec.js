import movies from "./before-after"

// --> Multiple duplicated function test is not best practice

/*
describe("My Favorite Movies", () => {
test("can add a movies", () => {
    const myMovies = [{
        title: "Terminator",
        rate: null
    }];

    movies.add(myMovies, "John Wick");
    expect(myMovies).toMatchSnapshot();
});

test("rate a movie", () => {
    const myMovies = [{
        title: "Terminator",
        rate: null
    }];

    movies.rate(myMovies[0], 5);
    expect(myMovies).toMatchSnapshot();
});


})
*/

describe('Favorite Movies', () => {
    let myMovies = []
    beforeEach(() => {
        myMovies = [{
        title: 'Terminator',
        rate: null
        }]
    })

    test('can add a movie', () => {
        movies.add(myMovies, 'John Wick')
        expect(myMovies).toMatchSnapshot()
    })

    test('rate a movie', () => {
        movies.rate(myMovies[0], 5)
        expect(myMovies).toMatchSnapshot()
    })
})