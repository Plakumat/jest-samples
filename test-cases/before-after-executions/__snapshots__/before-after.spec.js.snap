// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Movies can add a movie 1`] = `
Array [
  Object {
    "rate": null,
    "title": "Age of Ultron",
  },
  Object {
    "rate": null,
    "title": "Kung Fury",
  },
]
`;

exports[`Favorite Movies rate a movie 1`] = `
Array [
  Object {
    "rate": 5,
    "title": "Age of Ultron",
  },
]
`;
